### PR TITLE
User profile page

### PR DESCRIPTION
**Pull request by siviavio4ka**
- [x] The user's name must be displayed as a text block with an "Edit" button.
- [x] When the "Edit" button is clicked, the text block must turn into a form field to enter a new name, and the "Edit" button must change to a "Save" button.
- [x] When the "Save" button is clicked, the new name must be saved, the form field must revert to a text block, and the "Save" button must change back to an "Edit" button.
- [x] The user's email must be displayed as a text block with an "Edit" button.
- [x] When the "Edit" button is clicked, the text block must turn into a form field to enter a new email, and the "Edit" button must change to a "Save" button.
- [x] When the "Save" button is clicked, the new email must be saved, the form field must revert to a text block, and the "Save" button must change back to an "Edit" button.
- [x] The user profile page must include a "Logout" button.
- [x] When the "Logout" button is clicked, the user session must be terminated, and the user must be redirected to the main page.
- [x] The user profile page must include a "Change Password" button.
- [x] When the "Change Password" button is clicked, a modal window must open with one form field to enter the new password and a "Save" button.
- [x] When the "Save" button in the modal window is clicked, the new password must be saved, and the modal window must close upon successful password change.

### CODE REVIEW

File: src/app/pages/user-profile/user-profile.component.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  height: 56px;
+}
+
+.profile-info span {
+  font-size: 16px;
+}
+
+.profile-info button {
+  margin-left: 8px;
+}
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/user-profile.component.scss:34]: these classes are never used

File: src/app/pages/user-profile/user-profile.component.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    name: ['', Validators.required],
+    email: ['', [Validators.required, Validators.email]],
+  });
+
+  isEditName = false;
+  isEditEmail = false;
+
+  ngOnInit(): void {
+    this.profileService.loadProfile().subscribe((profile) => {
+      console.log(profile);
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/user-profile.component.ts:64]: do we need this console.log?

File: src/app/pages/user-profile/user-profile.component.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  toggleEditEmail() {
+    this.isEditEmail = !this.isEditEmail;
+  }
+
+  changePassword() {
+    const dialogRef = this.dialog.open(PasswordModalComponent);
+
+    dialogRef.afterClosed().subscribe((result) => {
+      console.log(result);
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/user-profile.component.ts:95]: console.log

File: src/app/pages/user-profile/user-profile.component.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      if (result) {
+        console.log(result);
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/user-profile.component.ts:97]: +1 console.log

File: src/app/pages/user-profile/password-modal/password-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import { Component, inject } from '@angular/core';
+import { NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatDialogModule, MatDialogRef } from '@angular/material/dialog';
+import { passwordValidation } from '../../../components/form-error-message/password-validation';
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/password-modal/password-modal.component.ts:4]:
```suggestion
import { passwordValidation } from '../../../validators/password-validation';
```
passwordValidation  were moved

File: src/app/pages/user-profile/password-modal/password-modal.component.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<h2 mat-dialog-title>Change Password</h2>
+<mat-dialog-content>
+  <form [formGroup]="passwordForm">
+    <mat-form-field appearance="outline" [style]="{padding: '10px'}">
+      <mat-label>New Password</mat-label>
+      <input matInput type="password" formControlName="password" />
+      <app-form-error-message formControlName="password"></app-form-error-message>
```
Comment (AYaki-coder) [inline src/app/pages/user-profile/password-modal/password-modal.component.html:7]:
```suggestion

      <mat-error><app-form-error-message formControlName="password"></app-form-error-message></mat-error>
```

**Review by AYaki-coder — CHANGES_REQUESTED** [review]

**Review by AYaki-coder — APPROVED** [review]

**Review by AV-Shell — APPROVED** [review]

**Pull request merged**